File: backend/src/main/java/br/com/calciolari/datahub/imports/domain/parser/ParsedImportStats.java
package br.com.calciolari.datahub.imports.domain.parser;

/**
 * Aggregate counts from a parse. Null means the metric was not observed / not
 * applicable yet — never invent zeros to look complete.
 */
public record ParsedImportStats(
		Integer pages,
		Integer lines,
		Integer uniqueSales,
		Integer entries,
		Integer exits
) {
	public static ParsedImportStats empty() {
		return new ParsedImportStats(null, null, null, null, null);
	}
}

File: backend/src/Calciolari.DataHub/Imports/Domain/Parser/ParsedImportStats.cs
namespace Calciolari.DataHub.Imports.Domain.Parser;

/// <summary>
/// Aggregate counts from a parse. Null means the metric was not observed / not
/// applicable yet — never invent zeros to look complete.
/// </summary>
public sealed record ParsedImportStats(
    int? Pages,
    int? Lines,
    int? UniqueSales,
    int? Entries,
    int? Exits)
{
    public static ParsedImportStats Empty { get; } = new(null, null, null, null, null);
}
